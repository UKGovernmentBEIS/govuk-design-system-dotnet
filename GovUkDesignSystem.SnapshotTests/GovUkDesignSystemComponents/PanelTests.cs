using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.GovUkDesignSystemComponents.Enums;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class PanelTests : SnapshotTestBase
    {
        private PanelViewModel DefaultPanelViewModel()
        {
            return new PanelViewModel
            {
                TitleText = "Title text",
                HeadingLevel = HeadingLevel.h1,
                Classes = "panel-class",
                Text = "Panel text"
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("Panel", DefaultPanelViewModel());
        }
        
        [Fact]
        public async Task Render_Html()
        {
            // Arrange
            var viewModel = DefaultPanelViewModel();
            viewModel.Text = null;
            viewModel.Html = o => "Html body";
            viewModel.TitleText = null;
            viewModel.TitleHtml = o => "Html title";
            
            // Act & Assert
            await VerifyPartial("Panel", viewModel);
        }
    }
}