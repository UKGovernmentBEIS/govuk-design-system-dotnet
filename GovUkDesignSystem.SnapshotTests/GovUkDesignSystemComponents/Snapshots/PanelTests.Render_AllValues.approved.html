<div class="govuk-panel govuk-panel--confirmation panel-class">
  <h1 class="govuk-panel__title">

Title text    </h1>
  <div class="govuk-panel__body">

Panel text    </div>
</div>